source/event_stream.c

```diff
@@ -749,37 +751,61 @@ struct aws_event_stream_header_value_pair aws_event_stream_create_int32_header(
     return header;
 }
 
-int aws_event_stream_add_byte_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int8_t value) {
+int aws_event_stream_add_bool_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int8_t value) {
+    struct aws_byte_cursor name_cursor = aws_byte_cursor_from_array(name, (size_t)name_len);
+
+    return aws_event_stream_add_bool_header_by_cursor(headers, name_cursor, value != 0);
+}
+
+int aws_event_stream_add_bool_header_by_cursor(
+    struct aws_array_list *headers,
+    struct aws_byte_cursor name,
+    bool value) {
+
     AWS_FATAL_PRECONDITION(headers);
-    AWS_FATAL_PRECONDITION(name);
+    AWS_FATAL_PRECONDITION(name.len > 0);
+    AWS_FATAL_PRECONDITION(name.ptr != NULL);
     AWS_RETURN_ERROR_IF(
-        name_len <= AWS_EVENT_STREAM_HEADER_NAME_LEN_MAX, AWS_ERROR_EVENT_STREAM_MESSAGE_INVALID_HEADERS_LEN);
+        name.len <= AWS_EVENT_STREAM_HEADER_NAME_LEN_MAX, AWS_ERROR_EVENT_STREAM_MESSAGE_INVALID_HEADERS_LEN);
```

Comment:
Minor question: Would it be possible to put these checks into a little helper function, since they are used for every function but are the same for each?